### ISSUE TITLE
HNT Rewards are not reliable in the CSV export

### ISSUE THREAD

**Issue by loggic-debug** [helium/explorer#312]
Hi Dear Helium-Team,

i have noticed that the HNT rewards are not the same as the actual mined amount. Probably it depends also to the downtime by end of last week.

what i have noticed is:
the helium.hiker.rocks shows me that i mined 96.* HNT from 1st to 7th April.
if i am going to export a CSV from the explorer of the hotspot or of the user wallet i only get like 74.*HNT in the datearea.

what i can also see is - that the graph of last 30 Days - shows also the amount of like 96.* HNTs - see screenshot. 

The CSV export would be used for taxes too i guess and this amount is pretty wrong like 30% distinguish. and the Hiker looks like it shows the right amount like the 30 Days graph. 

Please care also the probably second Bug in the Screenshots like 8.* HNT per day export - but hiker shows again the right amount but the explorer exports the 7th April too.

INFO: My Timezone is Bern/Switzerland

i hope that helps..... Thanks!

Best Regards,
Emre

![Unbenannt](https://user-images.githubusercontent.com/82143418/113991736-0b0c6000-9853-11eb-91a7-3964d217f7be.png)
![Unbenannt](https://user-images.githubusercontent.com/82143418/113991881-3000d300-9853-11eb-8281-cd48b548c262.png)
![Unbenannt](https://user-images.githubusercontent.com/82143418/113992236-8837d500-9853-11eb-8b51-f7520a615b4e.png)

![Unbenannt](https://user-images.githubusercontent.com/82143418/113991409-b10b9a80-9852-11eb-92b6-e491992bb09e.png)

**Comment by cokes518** [comment]
We've identified the bug and will be fixing it on the api side.

**Comment by mfalkvidd** [comment]
@cokes518 any details on the bug that can help us understand the difference?
Any progress on a fix?

We also see a difference of about 30%.

**Comment by danielcolinjames** [comment]
hi @mfalkvidd, could you provide some examples or more details about where you're seeing a discrepancy?

a difference of 30% between the CSV and what?

I believe the bug in this issue was because of a temporary glitch on the API side (which was fixed). but there may also be complications involving timezones / UTC and other things in the CSV export code that I haven't really had a chance to do a deep dive on.

so any specific examples or further details would be helpful :)

**Comment by mfalkvidd** [comment]
If I sum the transactions in the csv export, the current wallet balance is about 6k HNT. But the current wallet value according to explorer.helium.com is ~100 HNT.
I'd rather not associate myself with our HNT addresses in a public forum. I'll see if I can pick a random address and perform the same calculations.

**Comment by mfalkvidd** [comment]
Sorry for the delay. I have found the differences.

TLDR; csv file is missing oui and token_burn transactions

This is what I did:
* fetched all activity for the account from the API. That gave me a 16MB json file with 335,942 lines. Joy 😆 I used https://gist.github.com/mfalkvidd/b69d96727dd636a7c35c51d2de55093c
* compared the sum of all rewards in the json file with the sum of all the rewards in the csv file. *They match!*. one-liner: `jq '[.[].data[] | select(.type|startswith("rewards_v")) | .rewards[].amount]|add/100000000' all_activity.json`
* compared all payment transactions (v1 and v2). *They match!*

That led me to the realization that the entire difference was accounted for by the missing `oui_v1` and `token_burn_v1` from the csv file. I can fetch the token burn from the activity tab at https://explorer.helium.com/accounts/<account> but there is no easy wat to get oui_v1 transaction from explorer.helium.com. I can get the oui transaction from the API by using `jq '.[].data[] | select(.type=="oui_v1")' all_activity.json` though.

**Comment by cokes518** [comment]
We'll be retiring csv export in explorer v2 and directing users to community created tools.
